src/cdk/a11y/high-contrast-mode/high-contrast-mode-detector.ts

```diff
@@ -0,0 +1,89 @@
+ */
+
+import {Platform} from '@angular/cdk/platform';
+import {DOCUMENT} from '@angular/common';
+import {Inject, Injectable} from '@angular/core';
+
+
+/** Set of possible high-contrast mode backgrounds. */
+export enum HighContrastMode {
+  NONE,
+  BLACK_ON_WHITE,
+  WHITE_ON_BLACK,
+}
+
+/** CSS class applied to the document body when in black-on-white high-contrast mode. */
+export const BLACK_ON_WHITE_CSS_CLASS = 'cdk-high-contrast-black-on-white';
+
+/** CSS class applied to the document body when in white-on-black high-contrast mode. */
+export const WHITE_ON_BLACK_CSS_CLASS = 'cdk-high-contrast-white-on-black';
+
+/** CSS class applied to the document body when in high-contrast mode. */
+export const HIGH_CONTRAST_MODE_ACTIVE_CSS_CLASS = 'cdk-high-contrast-active';
+
+/**
+ * Service to determine whether the browser is currently in a high-constrast-mode environment.
+ *
+ * Microsoft Windows supports an accessibility feature called "High Contrast Mode". This mode
+ * changes the appearance of all applications, including web applications, to dramatically increase
+ * contrast.
+ */
+@Injectable({providedIn: 'root'})
+export class HighContrastModeDetector {
+  private _document: Document;
+
+  constructor(private _platfrom: Platform, @Inject(DOCUMENT) document: any) {
+    this._document = document;
+  }
+
+  /** Gets the current high-constrast-mode for the page. */
+  getHighContrastMode(): HighContrastMode {
+    if (!this._platfrom.isBrowser) {
+      return HighContrastMode.NONE;
+    }
+
+    // Create a test element with an arbitrary background-color that is neither black nor
+    // white; high-contrast mode will coerce the color to either black or white. Also ensure that
+    // appending the test element to the DOM does not affect layout by absolutely positioning it
+    const testElement = this._document.createElement('div');
+    testElement.style.backgroundColor = 'rgb(1,2,3)';
+    testElement.style.position = 'absolute';
+    this._document.body.appendChild(testElement);
+
+    // Get the computed style ofor the background color, collapsing spaces to normalize between
+    // browsers. Once we get this color, we no longer need the test element. Access the `window`
+    // via the document so we can fake it in tests.
+    const documentWindow = this._document.defaultView!;
+    const computedColor =
+        (documentWindow.getComputedStyle(testElement).backgroundColor || '').replace(/ /g, '');
+    this._document.body.removeChild(testElement);
+
+    switch (computedColor) {
+      case 'rgb(0,0,0)': return HighContrastMode.WHITE_ON_BLACK;
+      case 'rgb(1,1,1)': return HighContrastMode.BLACK_ON_WHITE;
+    }
+    return HighContrastMode.NONE;
+  }
+
+  /** Applies CSS classes indicating high-contrast mode to document body (browser-only). */
+  _applyBodyHighContrastModeCssClasses(): void {
+    if (this._platfrom.isBrowser) {
+      const body = this._document.body;
+      body.classList.remove(
```

Comment:
The signature of `classlist.remove` with multiple classes isn't supported on IE. You need to do 3 separate `remove` calls.

Reply:
Done